nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -128,15 +138,21 @@ bool BtActionServer<ActionT>::on_configure()
   client_node_->declare_parameter(
     "global_frame", node->get_parameter("global_frame").as_string());
 
+  // set the timeout in seconds for the action server to discard goal handles if not finished
+  int timeout;
+  node->get_parameter("action_server_result_timeout", timeout);
+  rcl_action_server_options_t server_options = rcl_action_server_get_default_options();
+  server_options.result_timeout.nanoseconds = RCL_S_TO_NS(timeout);
+
   action_server_ = std::make_shared<ActionServer>(
     node->get_node_base_interface(),
     node->get_node_clock_interface(),
     node->get_node_logging_interface(),
     node->get_node_waitables_interface(),
-    action_name_, std::bind(&BtActionServer<ActionT>::executeCallback, this));
+    action_name_, std::bind(&BtActionServer<ActionT>::executeCallback, this),
+    nullptr, std::chrono::milliseconds(500), false, server_options);
 
   // Get parameters for BT timeouts
-  int timeout;
```

Comment:
DONT reuse variables

Reply:
As I understand, the `timeout` is being already re-used later in the code for storing "bt_loop_duration" and "default_server_timeout", so one more parameter re-usage wouldn't be an issue, I believe. How about to leave the comment before each (re)usage to get it clear?

Reply:
Yikes. That's not good either. Lets make them all their own variables.